### PR TITLE
React capstone project - Covid Metrics webapp

### PR DESCRIPTION
**Pull request by jeanbuhendwa**
#### In this pull request, I have : 

- [ ] Selected the API  that I have used (Covid API).
- [ ] Built file structure for the React app.
- [ ] Prepared routes and navigation in the app.
- [ ] Displayed a list of items and filter them. Data come from the API.
- [ ] Created the tests for the application.
- [ ] Styled the components to match the design provided.
- [ ] Deployed the project [Live demo](https://644130fd0eeb5f22bec1933b--genuine-faloodeh-10b92c.netlify.app)
- [ ] Recorded a video for the project. [Video Outlines](https://www.loom.com/share/7a07ff668a6a41a3ae270572af911b39)
- [ ] Created a good README and PR description.

### CODE REVIEW

File: src/__test__/App.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
+import NavBar from '../components/navbar';
+import '@testing-library/jest-dom';
+
+test('renders navbar without errors', () => {
```
Comment (rustam-tolipov) [inline src/__test__/App.test.js:7]: **You did a great job! I like the clean code that you wrote!**

- [ ] However, I only can see one test file in your project that is related to the unit test. But, students are meant to write integration tests too. Please implement integration tests in your code. Thank you!

> HINT: You can perform a mocking data fetching and Redux Store connection.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @jeanbuhendwa,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Great design_
_Clean code_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/App.test.js (hunk 1 above)
Comment (jeanbuhendwa) [reply src/__test__/App.test.js:7]: Thanks @Rustamxon7 for the feedback. I will work on it 😎

File: src/__test__/navbar.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  afterEach(() => {
+    mockUseLocation.mockRestore();
+  });
+
+  test("renders without errors", () => {
+    render(
+      <BrowserRouter>
+        <NavBar />
+      </BrowserRouter>
+    );
```
Comment (pinkmoon25) [inline src/__test__/navbar.test.js]: - [ ] Good work so far, however as mentioned by the previous reviewer https://github.com/jeanbuhendwa/covidApp/pull/6#discussion_r1171229283 please add more test cases, as required in the project too :point_down: 
image

currently there is only test case for the navbar component, please add test case for the homepage and detailspage component, and you can also add tests for checking the reducer by providing a predefined state and action and expecting the reducer to behave accordingly,

**Review by pinkmoon25 — CHANGES_REQUESTED** [review]
Hi @jeanbuhendwa , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
Linter checks :heavy_check_mark: 
gitflow :heavy_check_mark: 
react and redux :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/homepage.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                  {' '}
+                  <span className="pop-span">{country.population}</span>
+                </p>
+              </div>
+            </li>
+          </Link>
+        ))}
+      </ul>
+    </>
+  );
```
Comment (mckent05) [inline src/pages/homepage.js:54]: **OPTIONAL** - Good job done so far, but according to the project requirements, `The Home page has a list of categories that could be filtered using given parameters, e.g. category name` in your case by `country name`. Please fix this by including a feature that filters the list of countries 🙏

**Review by mckent05 — APPROVED** [review]
Hi @jeanbuhendwa, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏
- All tests are passing ✔️ ✔️  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**